stem/directory.py

```diff
@@ -383,7 +384,7 @@ def from_cache(path: str = FALLBACK_CACHE_PATH) -> Dict[str, 'stem.directory.Fal
       results[fingerprint] = Fallback(
         address = attr['address'],
         or_port = int(attr['or_port']),
-        dir_port = int(attr['dir_port']),
+        dir_port = int(attr['dir_port']) if attr['dir_port'] else None,
```

Comment:
I'm not sure about this one, it wasn't in the commit you proposed, but wouldn't we have the same issue here?